libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -527,6 +528,34 @@ private boolean isEvidenceVc(VcStoreItem item) throws CredentialParseException {
         return false;
     }
 
+    private boolean isNonZeroInt(JsonNode node) {
+        return node.isInt() && node.asInt() != 0;
+    }
+
+    private boolean isValidEvidence(VcStoreItem item) {
+        try {
+            JsonNode vcEvidenceNode = getVCClaimNode(item.getCredential(), VC_EVIDENCE);
+            return ((vcEvidenceNode.size() > 0)
+                    && isNonZeroInt(vcEvidenceNode.get(0).path(VC_EVIDENCE_VALIDITY))
+                    && isNonZeroInt(vcEvidenceNode.get(0).path(VC_EVIDENCE_STRENGTH)));
+        } catch (CredentialParseException e) {
+            return false;
+        }
+    }
+
+    public String getCredentialIssuerIfSingleValidEvidence(List<VcStoreItem> vcStoreItems) {
+        Map<String, Long> issuerCountMap =
+                vcStoreItems.stream()
+                        .filter(item -> isValidEvidence(item))
+                        .map(item -> item.getCredentialIssuer())
+                        .collect(Collectors.groupingBy(issuer -> issuer, Collectors.counting()));
+        if (issuerCountMap.size() == 1
+                && issuerCountMap.values().stream().anyMatch(count -> count >= 1)) {
```

Comment:
This doesn't match the name of the function - you could have multiple bits of evidence from the same issuer and that issuer would be returned.